scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulerFrontendState.java

```diff
@@ -1245,12 +1241,29 @@ public UserData getCurrentUserData() throws NotConnectedException {
         return PASchedulerProperties.getPropertiesAsHashMap();
     }
 
+    synchronized ClientJobState getClientJobState(JobId jobId) {
+        if (!jobsMap.containsKey(jobId)) {
+            List<InternalJob> internalJobs = dbManager.loadInternalJob(Arrays.asList(jobId.longValue()));
+            if (!internalJobs.isEmpty()) {
+                InternalJob internalJob = internalJobs.get(0);
+                ClientJobState clientJobState = new ClientJobState(internalJob);
+                jobsMap.put(jobId, clientJobState);
```

Comment:
Beware that by doing jobsMap.put(), the job will be kept in memory without housekeeping